ra-optimisation/linear-rao/src/main/java/com/farao_community/farao/linear_rao/LinearRao.java

```diff
@@ -122,19 +123,23 @@ public String getVersion() {
             LOGGER.info("Iteration {} - systematic analysis [end]", iteration);
 
             if (linearRaoData.getCracResult(optimizedVariantId).getCost() < linearRaoData.getCracResult(bestVariantId).getCost()) { // if the solution has been improved, continue the search
-                LOGGER.warn("Iteration {} - Better solution found with a minimum margin of {} MW", iteration, -linearRaoData.getCracResult(optimizedVariantId).getCost());
+
+                LOGGER.warn(String.format("Iteration %d - Better solution found with a minimum margin of %.2f %s (optimisation criterion : %.2f)",
+                    iteration, -linearRaoData.getCracResult(optimizedVariantId).getFunctionalCost(), linearRaoParameters.getObjectiveFunctionUnit(), linearRaoData.getCracResult(optimizedVariantId).getCost()));
+
                 if (!bestVariantId.equals(linearRaoData.getInitialVariantId())) {
                     linearRaoData.deleteVariant(bestVariantId, false);
                 }
+
                 bestVariantId = optimizedVariantId;
             } else { // unexpected behaviour, stop the search
-                LOGGER.warn("Iteration {} - Linear Optimization found a worse result than previous iteration: from {} MW to {} MW",
-                    iteration, -linearRaoData.getCracResult(bestVariantId).getCost(), -linearRaoData.getCracResult(optimizedVariantId).getCost());
+                LOGGER.warn(String.format("Iteration %d - Linear Optimization found a worse result than previous iteration, from %.2f to %.2f (optimisation criterion)",
+                    iteration, linearRaoData.getCracResult(bestVariantId).getCost(), linearRaoData.getCracResult(optimizedVariantId).getCost()));
```

Comment:
Here maybe we could log if the linear optimization found a worst result because of the invalidity of the linear approximation or because of something else (switch to fallback, loopflow violation...)

Reply:
I added the information in this log on the evolution of the functional cost vs. overall cost